src/amo/components/Category/index.js

```diff
@@ -32,6 +33,10 @@ export class CategoryBase extends React.Component {
   render() {
     const { category, ...searchProps } = this.props;
 
+    if (!category) {
```

Comment:
`category` will also be `null` if this page is loading, which will happen with a saga. We should be proactive and make this handle both the loading case and the empty case.

Looks like the [categories reducer](https://github.com/mozilla/addons-frontend/blob/master/src/core/reducers/categories.js#L37) adds a loading state we can use to check if the page is loading or has already loaded. If `loading` is `false` and `category` is false-y then we can return `<NotFound />`; otherwise this will 404 during category loads.